packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -129,13 +144,26 @@ const Heatmap = React.forwardRef(function Heatmap(
   const clipPathId = `${id}-clip-path`;
 
   const defaultizedXAxis = React.useMemo(
-    () => xAxis.map((axis) => ({ scaleType: 'band' as const, categoryGapRatio: 0, ...axis })),
-    [xAxis],
+    () =>
+      (xAxis && xAxis.length > 0 ? xAxis : [{}]).map((axis) => ({
+        id: DEFAULT_X_AXIS_KEY,
```

Comment:
It's nitpicking because heatmap should not have multiple axes.

But this would ensure a difference between `id: DEFAULT_X_AXIS_KEY` which is an axis the user did nothing for. And the `id: defaultized-x-axis-0` which is an axis provided by the user but without specifying its id.

```suggestion
      (xAxis && xAxis.length > 0 ? xAxis : [{ id: DEFAULT_X_AXIS_KEY }]).map((axis) => ({
```